src/ReferenceFEs/DuffyQuadratures.jl

```diff
@@ -52,22 +52,24 @@ end
 
 _duffy_map(q::Point{1,T}) where T = q
 
-function _gauss_jacobi_in_0_to_1(order,alpha,beta)
+function _gauss_jacobi_in_0_to_1(order,alpha,beta;fptype::Type{<:AbstractFloat}=Float64)
   n = _npoints_from_order(order)
   x,w = gaussjacobi(n,alpha,beta)
-  _map_to(0,1,x,w)
+  _map_to(0,1,x,w;T=fptype)
 end
 
-function _gauss_legendre_in_0_to_1(order)
+function _gauss_legendre_in_0_to_1(order;fptype::Type{<:AbstractFloat}=Float64)
   n = _npoints_from_order(order)
   x,w = gausslegendre(n)
-  _map_to(0,1,x,w)
+  _map_to(0,1,x,w;T=fptype)
 end
 
 # Transforms a 1-D quadrature from `[-1,1]` to `[a,b]`, with `a<b`.
-function _map_to(a,b,points,weights)
-  points_ab = 0.5*(b-a)*points .+ 0.5*(a+b)
-  weights_ab = 0.5*(b-a)*weights
+function _map_to(a,b,points,weights;T::Type{<:AbstractFloat}=Float64)
+  points_ab = Vector{T}(undef,length(points))
+  weights_ab = Vector{T}(undef,length(weights))
+  points_ab .= 0.5*(b-a)*points .+ 0.5*(a+b)
```

Comment:
Here we might be mixing different floating point precision numbers.  

E.g. `0.5` is `Float64` while `points_ab` eltype might be, say, `Float32`. 

Thus, a number of implicit conversions are being performed in that case. 

I wonder which is the performance impact of these conversions and whether it is convenient to be systematic in performing conversions so that we do not have many implicit conversions.

Reply:
You are right! I felt implicit conversion is more efficient than first forming the Vector in `Float64` (as the values coming from `gausslegendre` and `gaussjacobi` are of `Float64` eltype) and then constructing a new vector of  converted `Float32` values. To handle `0.5` I can put `T` around it, if you also think it would be good idea.

Reply:
I tested a bit on such situations and it doesn't seem necessary to convert the RHS to the floating point type of LHS and the implicit conversion is slightly faster. I added places where the usage of stated floating precision for LHS was missing (very minor ones actually). I think it is better to have consistent floating point precision for elements in RHS (doesn't have to be the same as LHS if implicit conversion is possible!)

Reply:
> **it doesn't seem necessary** to convert the RHS to the floating point type of LHS

You mean explicit conversion, correct? I indeed see the need to perform the conversion (no matter whether it is generated by the compiler implicitly or by the programmer explicitly) if the `eltype`s of both sides do not match.

Something we could also do is to force that `a`, `b`, `points` and `weights` are of the same type and perform the conversion (explicitly) before calling the function. This way you do not have to pass optional `T`. What do you think?

Reply:
> (as the values coming from gausslegendre and gaussjacobi are of Float64 eltype)

We would extend `gausslegendre` and `gaussjacobi` with a new method each that takes an extra argument with `fptype`  and internally calls the current `gausslegendre` and `gaussjacobi` and performs the explicit conversion.

Reply:
> You mean explicit conversion, correct? I indeed see the need to perform the conversion (no matter whether it is generated by the compiler implicitly or by the programmer explicitly) if the eltypes of both sides do not match.

yes true.. I meant explicit conversion. 

> Something we could also do is to force that a, b, points and weights are of the same type and perform the conversion (explicitly) before calling the function. This way you do not have to pass optional T. What do you think?

This could be done but this incurs additional memory to store the new vectors of converted type, anyways I feel since `a`, `b`, `points` and `weights` are all in `Float64`, the RHS happens consistently in `Float64` without needing conversion until the storage into LHS, this helps in not having internal floating point cancellations in RHS calculation due to a lower precision arithmetic.

Reply:
> We would extend gausslegendre and gaussjacobi with a new method each that takes an extra argument with fptype and internally calls the current gausslegendre and gaussjacobi and performs the explicit conversion.

Like mentioned above this might lead to degraded prec even at the RHS calculation stage, downgrading to lower precision can be done only at the storage point, anyways the scaling (which is happening in the RHS here) doesn't bring in much computation difference even done in `Float32` as it is being done only once when quadrature is setup, and might lead to lower accuracy due to internal cancellations, I feel.

Reply:
Ok, I think the argument of degraded precision might be a good one. The strategy would be to perform the computations leading to the final weights and points in `Float64`, and defer the conversion to the very last moment, i.e., when you have to store the result into the corresponding arrays of the desired precision. But isn't this contradictory with the following you did?

> on the other hand gauss from QuadGK.jl has this option and I used it.

Reply:
True, rightly pointed out the inconsistency, I aware of this but even then I did this to avoid extra storage for conversion and also because the manipulation after the gauss points and weights are generated is very little and mostly trivial (`1.0` and `0.5` have exact storage in both precision!)

Reply:
Yes Alberto..

Reply:
> Yes Alberto..

ok, perfect.